app/src/main/java/fr/gaulupeau/apps/Poche/ui/ReadArticleActivity.java

```diff
@@ -337,50 +344,43 @@ public boolean onOptionsItemSelected(MenuItem item) {
 
     @Override
     public boolean dispatchKeyEvent(KeyEvent event) {
-
-        // only do this once per key press (otherwise the action is done twice)
         if (event.getAction() == KeyEvent.ACTION_DOWN) {
-            switch (event.getKeyCode()) {
+            int code = event.getKeyCode();
+            switch (code) {
                 case KeyEvent.KEYCODE_PAGE_UP:
-                    scroll(true, screenScrollingPercent, smoothScrolling);
-                    return true;
-
                 case KeyEvent.KEYCODE_PAGE_DOWN:
-                    scroll(false, screenScrollingPercent, smoothScrolling);
-                    return true;
-
-                case KeyEvent.KEYCODE_CAMERA:
-                    disableTouch = !disableTouch;
-                    if (disableTouch) {
-                        Toast.makeText(this, "Touch screen disabled", Toast.LENGTH_SHORT).show();
-                    } else {
-                        Toast.makeText(this, "Touch screen enabled", Toast.LENGTH_SHORT).show();
-                    }
-                    Log.d(TAG, "toggling touch screen, now disableTouch is " + disableTouch);
+                    scroll(code == KeyEvent.KEYCODE_PAGE_UP, screenScrollingPercent, smoothScrolling);
                     return true;
 
                 case KeyEvent.KEYCODE_VOLUME_UP:
-                    if (volumeButtonsScrolling) {
-                        scroll(true, screenScrollingPercent, smoothScrolling);
-                        return true;
-                    } else {
-                        return super.dispatchKeyEvent(event);
-                    }
-
                 case KeyEvent.KEYCODE_VOLUME_DOWN:
                     if (volumeButtonsScrolling) {
-                        scroll(false, screenScrollingPercent, smoothScrolling);
+                        scroll(code == KeyEvent.KEYCODE_VOLUME_UP, screenScrollingPercent, smoothScrolling);
                         return true;
-                    } else {
-                        return super.dispatchKeyEvent(event);
                     }
+                    break;
+            }
+
+            if(event.getKeyCode() == disableTouchKeyCode && (disableTouch || disableTouchOptionEnabled)) {
+                disableTouch = !disableTouch;
 
-            } // end switch
-        } // end if
+                Log.d(TAG, "toggling touch screen, now disableTouch is " + disableTouch);
+                Toast.makeText(this, disableTouch
+                                ? R.string.message_disableTouch_inputDisabled
+                                : R.string.message_disableTouch_inputEnabled,
+                        Toast.LENGTH_SHORT).show();
+                return true;
+            }
+        }
 
         return super.dispatchKeyEvent(event);
     }
 
+    @Override
+    public boolean dispatchTouchEvent(MotionEvent ev) {
+        return disableTouch || super.dispatchTouchEvent(ev);
```

Comment:
just out of curiosity and to learn android stuff: should this function not return false if it didn't use the event? or is it legitimate to "just" call the `super.dispatchTouchEvent()` and rely on a proper return code?

Reply:
It is not really Android stuff, but general extension logic. We **add** behaviour (as opposed to replacing it completely), so, in this case we can consume event (`return disableTouch`) or let the default implementation do its work (` || super.dispatchTouchEvent(ev)`). If we didn't override this method, the default (`super.dispatchTouchEvent(ev)`) implementation would be called directly, so we kinda presume it returns proper values.
I hope it helps.